test/unit/org/apache/cassandra/cql3/validation/entities/CollectionsTest.java

```diff
@@ -1980,4 +1981,99 @@ public void testSelectionOfEmptyCollections() throws Throwable
             assertRows(execute("SELECT m['0'..'1']['3'..'5'], s[0..1][3..5] FROM %s WHERE k = 3"), row(null, null));
         });
     }
+
+    // Tests for CASSANDRA-17623
+    // Before CASSANDRA-17623, parameterized queries with maps as values would fail because frozen maps were
+    // required to be sorted by the sort order of their key type, but weren't always sorted correctly.
+    // Also adding tests for Sets, which did work because they always used SortedSet, to make sure this behavior is maintained.
+    // We use `executeNet` in these tests because `execute` passes parameters through CqlTester#transformValues(), which calls
+    // AbstractType#decompose() on the value, which "fixes" the map order, but wouldn't happen normally.
```

Comment:
Nit: this could be a block comment:
```suggestion
    /*
     Tests for CASSANDRA-17623.
     Before CASSANDRA-17623, parameterized queries with maps as values would fail because frozen maps were required to 
     be sorted by the sort order of their key type, but weren't always sorted correctly. Also adding tests for Sets, 
     which did work because they always used SortedSet, to make sure this behavior is maintained.
     We use `executeNet` in these tests because `execute` passes parameters through CqlTester#transformValues(), which 
     calls AbstractType#decompose() on the value, which "fixes" the map order, but wouldn't happen normally.
     */

```